### ISSUE TITLE
Remove FlowFuture in favour of Runnable callback

### ISSUE THREAD

**Issue by sagenschneider** [sagenschneider/OfficeFloor#4]
FlowFuture requires synchronising of state.  It also enables multiple joins that need to be accounted for (when typically only the one is used, which is the invoker of the flow).

Rather than Flow Future and Join, provide optional Runnable callback to doFlow(...) method that is invoked in another task once the flow is complete.

Note the same can be undertaken for invokeProcess to provide a Runnable to execute once the Process Completes.

**Comment by sagenschneider** [comment]
Implemented in v3
